lib/jna/clink_jna.cc

```diff
@@ -26,20 +27,30 @@
 extern "C" {
 #endif
 
+tfrt::HostContext *jna_host_context =
+    clink::CreateHostContext("mstd", tfrt::HostAllocatorType::kLeakCheckMalloc)
+        .release();
+
 // Clink JNA programs handle exception in Java. C++ part of the programs only
```

Comment:
The purpose of `CLINK_JNA_HANDLE_ERROR` and `CLINK_JNA_HANDLE_ASYNC_ERROR` is to record error information somewhere so that users can use this information to debug. As our code uses Java and C++, in order to have a full trace of the error we need to record the error on both sides. On the Java side we have been throwing `LastErrorException`, and Java handles the exception stack well. On the C++ side there is no proper infrastructure for now, so I added `CLINK_JNA_HANDLE_ERROR` and `CLINK_JNA_HANDLE_ASYNC_ERROR`.

Before adding more document to these macros, I began to realize that these are premature solutions to log recording and error tracking. So instead of adding document to it, shall we just remove these macros and just set `errno` for now, and add a TODO or issue to solve this problem afterwards?

Reply:
Per offline discussion, let's make sure to wait for AsyncValue to be ready before getting and returning its value to the Java caller. And let's document the behavior properly.